### DISCUSSION TITLE
How to minimize Main Thread Work to Improve Core Web Vitals

### DISCUSSION THREAD

**Discussion by Mathie01** [vercel/next.js#64907] [Help]
### Summary

Hello,

I have a good Web vitals score but the problem is that I still have a problem with "Reduce the workload of the main thread" which is very high.

My website is 100% static (SSG): I don't retrieve any data from a server or other

How to improve this?

Thanks!

![image](https://github.com/vercel/next.js/assets/67562521/0244393e-ed7c-4c59-b32d-69501a35da8c)

**Comment by icyJoseph** [comment]
How are you generating this result? CPU slowdown?

**Reply by Mathie01** [reply]
I'm using this website : https://pagespeed.web.dev/

**Reply by icyJoseph** [reply]
Right, they do generated a shareable trace and stuff, that'd be nice to look at. Other than that, are you reporting webvitals? Often you can see actual user data in pagespeed.

The one thing that might be blocking right now, is hydration. Can you profile your app in dev mode and try to see if there's __too__ many re-renders?

**Reply by Mathie01** [reply]
> Right, they do generated a shareable trace and stuff, that'd be nice to look at. Other than that, are you reporting webvitals? Often you can see actual user data in pagespeed.

I've already improved the score a lot. I haven't got any new user data yet. I'll keep an eye on it!

> The one thing that might be blocking right now, is hydration. Can you profile your app in dev mode and try to see if there's too many re-renders?

I don't have any re-renders!
What's strange is that almost everything is in static files, so I don't understand why there's so much JS.

My website : https://skezia.io/fr/

**Reply by icyJoseph** [reply]
I think the problem is that you download 18.2kb of CSS, render blocking, but according to pagespeed, you can drop 18.1kb of that... What's up with that `all.min.css` that you use?

**Reply by Mathie01** [reply]
It's FontAwesome! I've just removed it but I still have the same problem (I no longer have the alert on the report)

**Reply by icyJoseph** [reply]
Nice. I wonder what's inside this chunk, /_next/static/chunks/359-56fdb3d16ece0a52.js, it looks like about half of the file goes unused. You can use, https://www.npmjs.com/package/@next/bundle-analyzer, to create a map of bundles yourself, and see what is getting accidentally added there.

**Reply by Mathie01** [reply]
The map of my app :

[client.zip](https://github.com/vercel/next.js/files/15096684/client.zip)

**Reply by icyJoseph** [reply]
Could you just share a screenshot? :)

**Reply by Mathie01** [reply]
Yes I can !

![Capture d'écran 2024-04-25 105745](https://github.com/vercel/next.js/assets/67562521/fce9761d-4fda-477b-8a31-4056051655c9)

**Reply by icyJoseph** [reply]
Alright, so I'd say, drop axios first, and perhaps react-scroll as well. Fetch has wide support and there are alternatives that are not as taxing as axios is, https://bundlephobia.com/package/axios@1.6.8 

Maybe you are not even using it client side? So that's a good indicator to figure out why axios is accidentally ending up in the client bundle.

**Reply by Mathie01** [reply]
I don't know if this will improve the situation much: I use axios on the client side!
